light/rpc/client_test.go

```diff
@@ -1,90 +1,76 @@
-		Spec:  ics23.IavlSpec,
-		Key:   key,
-		Proof: commitmentProof,
-	}
-
-	next := &rpcmock.Client{}
-	next.On(
-		"ABCIQueryWithOptions",
-		context.Background(),
-		mock.AnythingOfType("string"),
-		bytes.HexBytes(key),
-		mock.AnythingOfType("client.ABCIQueryOptions"),
-	).Return(&ctypes.ResultABCIQuery{
-		Response: abci.ResponseQuery{
-			Code:   0,
-			Key:    key,
-			Value:  value,
-			Height: 1,
-			ProofOps: &tmcrypto.ProofOps{
-				Ops: []tmcrypto.ProofOp{op.ProofOp()},
-			},
-		},
-	}, nil)
-
-	lc := &lcmock.LightClient{}
-	appHash, _ := hex.DecodeString("5EFD44055350B5CC34DBD26085347A9DBBE44EA192B9286A9FC107F40EA1FAC5")
-	lc.On("VerifyLightBlockAtHeight", context.Background(), int64(2), mock.AnythingOfType("time.Time")).Return(
-		&types.LightBlock{
-			SignedHeader: &types.SignedHeader{
-				Header: &types.Header{AppHash: appHash},
-			},
-		},
-		nil,
-	)
-
-	c := NewClient(next, lc,
-		KeyPathFn(func(_ string, key []byte) (merkle.KeyPath, error) {
-			kp := merkle.KeyPath{}
-			kp = kp.AppendKey(key, merkle.KeyEncodingURL)
-			return kp, nil
-		}))
-	c.RegisterOpDecoder("ics23:iavl", testOpDecoder)
-	res, err := c.ABCIQuery(context.Background(), "/store/accounts/key", key)
-	require.NoError(t, err)
-
-	assert.NotNil(t, res)
-}
+// func TestABCIQuery(t *testing.T) {
+// 	tree, err := iavl.NewMutableTree(dbm.NewMemDB(), 100)
+// 	require.NoError(t, err)
+
+// 	var (
+// 		key   = []byte("foo")
+// 		value = []byte("bar")
+// 	)
+// 	tree.Set(key, value)
+
+// 	commitmentProof, err := tree.GetMembershipProof(key)
+// 	require.NoError(t, err)
+
+// 	op := &testOp{
+// 		Spec:  ics23.IavlSpec,
+// 		Key:   key,
+// 		Proof: commitmentProof,
+// 	}
+
+// 	next := &rpcmock.Client{}
+// 	next.On(
+// 		"ABCIQueryWithOptions",
+// 		context.Background(),
+// 		mock.AnythingOfType("string"),
+// 		bytes.HexBytes(key),
+// 		mock.AnythingOfType("client.ABCIQueryOptions"),
+// 	).Return(&ctypes.ResultABCIQuery{
+// 		Response: abci.ResponseQuery{
+// 			Code:   0,
+// 			Key:    key,
+// 			Value:  value,
+// 			Height: 1,
+// 			ProofOps: &tmcrypto.ProofOps{
+// 				Ops: []tmcrypto.ProofOp{op.ProofOp()},
+// 			},
+// 		},
+// 	}, nil)
+
+// 	lc := &lcmock.LightClient{}
+// 	appHash, _ := hex.DecodeString("5EFD44055350B5CC34DBD26085347A9DBBE44EA192B9286A9FC107F40EA1FAC5")
+// 	lc.On("VerifyLightBlockAtHeight", context.Background(), int64(2), mock.AnythingOfType("time.Time")).Return(
+// 		&types.LightBlock{
+// 			SignedHeader: &types.SignedHeader{
+// 				Header: &types.Header{AppHash: appHash},
+// 			},
+// 		},
+// 		nil,
+// 	)
+
+// 	c := NewClient(next, lc,
+// 		KeyPathFn(func(_ string, key []byte) (merkle.KeyPath, error) {
+// 			kp := merkle.KeyPath{}
+// 			kp = kp.AppendKey(key, merkle.KeyEncodingURL)
+// 			return kp, nil
+// 		}))
+// 	c.RegisterOpDecoder("ics23:iavl", testOpDecoder)
+// 	res, err := c.ABCIQuery(context.Background(), "/store/accounts/key", key)
+// 	require.NoError(t, err)
+
+// 	assert.NotNil(t, res)
+// }
```

Comment:
What is the rationale behind commenting this out vs. deleting the code? 

@evan-forbes can you add an equivalent test into the app repo?

Reply:
Wanted to leave it around as an example. 

P.S. The sdk doenst have the notion of a light client, so it first needs to be added (https://github.com/cosmos/cosmos-sdk/issues/6563)

Reply:
Yeah, I think removing makes sense. We can link to the still existing code in the tendermint repo in an issue (over in the app repo or here). Thanks for the context regarding the light client 🙏🏼